src/vcpkg/vcpkgcmdarguments.cpp

```diff
@@ -260,6 +260,11 @@ namespace vcpkg
                 }
                 continue;
             }
+            if (basic_arg == "--version" && args.command.empty())
```

Comment:
I'd suggest moving this up to the top as:
```cpp
if (arg_first != arg_last && arg_first + 1 == arg_last && *arg_first == "--version") {
    args.command = "version";
    return args;
}
```

Reply:
Yeah that sounds better